src/main/java/com/oltpbenchmark/benchmarks/tpch/procedures/GenericQuery.java

```diff
@@ -34,12 +35,18 @@ public abstract class GenericQuery extends Procedure {
     protected abstract PreparedStatement getStatement(Connection conn, RandomGenerator rand, double scaleFactor) throws SQLException;
 
     public void run(Connection conn, RandomGenerator rand, double scaleFactor) throws SQLException {
-
-        try (PreparedStatement stmt = getStatement(conn, rand, scaleFactor); ResultSet rs = stmt.executeQuery()) {
-            while (rs.next()) {
-                //do nothing
+        try (PreparedStatement stmt = getStatement(conn, rand, scaleFactor)) {
+            try (ResultSet rs = stmt.executeQuery()) {
```

Comment:
Splitting these made it slightly easier to see which line the exception was coming from, but it's probably not necessary.